modules/exadsBidAdapter.js

```diff
@@ -0,0 +1,433 @@
+  const videoInfo = bid.mediaTypes?.video;
+  const isValid = !!(
+    bid.params.endpoint &&
+    bid.params.userIp &&
+    bid.params.hasOwnProperty('userId') &&
+    bid.params.zoneId &&
+    bid.params.partner &&
+    bid.params.fid &&
+    bid.params.siteId &&
+    bid.params.impressionId &&
+    bid.params.country &&
+    bid.params.country.length > 0 &&
+    (bannerInfo || nativeInfo || videoInfo) &&
+    !!(bid.params.bidfloor && bid.params.bidfloorcur) &&
+    (nativeInfo ? bid.params.native &&
+      nativeInfo.ortb.assets &&
+      nativeInfo.ortb.assets.some(asset => !!asset.img) : true) &&
+    (videoInfo ? !!(bid.params.stream &&
+      bid.params.stream.video &&
+      bid.params.stream.video.mimes &&
+      bid.params.stream.video.mimes.length > 0 &&
+      bid.params.stream.protocols &&
+      bid.params.stream.protocols.length > 0) : true));
+
+  if (!isValid) {
+    utils.logError('Validation Error');
+  }
+
+  return isValid;
+}
+
+function hasValue(value) {
+  return (
+    value !== undefined &&
+    value !== null
+  );
+}
+
+function getGdprConsentChoice(bidderRequest) {
+  const hasGdprConsent =
+    hasValue(bidderRequest) &&
+    hasValue(bidderRequest.gdprConsent);
+
+  if (hasGdprConsent) {
+    return bidderRequest.gdprConsent;
+  }
+
+  return null;
+}
+
+export const spec = {
+  aliases: ['exads'], // short code
+  supportedMediaTypes: [BANNER, NATIVE, VIDEO],
+  isBidRequestValid: function (bid) {
+    utils.logInfo('on isBidRequestValid -> bid:', bid);
+
+    if (bid.bidder !== BIDDER) {
+      utils.logError('Validation Error', 'bidder wrong');
+      return false;
+    } else if (!bid.params.partner) {
+      utils.logError('Validation Error', 'bid.params.partner missed');
+      return false;
+    } else if (!Object.values(PARTNERS).includes(bid.params.partner)) {
+      utils.logError('Validation Error', 'bid.params.partner is not valid');
+      return false;
+    }
+
+    let adPartner = bid.params.partner;
+
+    if (adPartnerHandlers[adPartner] && adPartnerHandlers[adPartner]['validation']) {
+      return adPartnerHandlers[adPartner]['validation'](bid);
+    } else {
+      // Handle unknown or unsupported ad partners
+      return false;
+    }
+  },
+  buildRequests: function (validBidRequests, bidderRequest) {
+    utils.logInfo('on buildRequests -> validBidRequests:', validBidRequests);
+    utils.logInfo('on buildRequests -> bidderRequest:', bidderRequest);
+
+    return validBidRequests.map(bid => {
+      let adPartner = bid.params.partner;
+
+      imps.set(bid.params.impressionId, adPartner);
+
+      let endpointUrl = getUrl(adPartner, bid);
+
+      // Call the handler for the ad partner, passing relevant parameters
+      if (adPartnerHandlers[adPartner]['request']) {
+        return adPartnerHandlers[adPartner]['request'](bid, endpointUrl, validBidRequests, bidderRequest);
+      } else {
+        // Handle unknown or unsupported ad partners
+        return null;
+      }
+    });
+  },
```

Comment:
Looking at `outbrainBidAdapter` or `rubiconBidAdapter` adapters, they use this section to enrich the information returned like for example:

```
      ext: {
        prebid: {
          channel: {
            name: 'pbjs',
            version: '$prebid.version$'
          }
        }
      }
```

We could consider doing the same.

Reply:
Good idea. It could be good for tracking purposes on BE side. But BE should use it.

On my side I've just added it. 

Now we will have a similar request (it will change based on prebidJS version):

`{
    "sub": 0,
    "prebid": {
        "channel": {
            "name": "pbjs",
            "version": "7.25.0-pre"
        }
    }
}`